components/eamxx/scripts/test_all_scream.py

```diff
@@ -129,34 +141,126 @@ def __init__(self):
+
+    def __init__(self, _):
+        TestProperty.__init__(
+            self,
+            "valgrind",
+            "debug with valgrind",
+            [("CMAKE_BUILD_TYPE", "Debug"), ("EKAT_ENABLE_VALGRIND", "True")],
+            uses_baselines=False,
+            on_by_default=False,
+            default_test_len="short"
+        )
+
+###############################################################################
+class CMC(TestProperty):
+###############################################################################
+
+    def __init__(self, _):
+        TestProperty.__init__(
+            self,
+            "cuda_mem_check",
+            "debug with cuda memcheck",
+            [("CMAKE_BUILD_TYPE", "Debug"), ("EKAT_ENABLE_CUDA_MEMCHECK", "True")],
+            uses_baselines=False,
+            on_by_default=False,
+            default_test_len="short"
         )
 
 ###############################################################################
-def test_factory(user_req_tests, machine, mem_check):
+class CSM(TestProperty):
+###############################################################################
+
+    def __init__(self, _):
+        TestProperty.__init__(
+            self,
+            "compute_santizer_memcheck",
+            "debug with compute sanitizer memcheck",
+            [("CMAKE_BUILD_TYPE", "Debug"), ("EKAT_ENABLE_COMPUTE_SANITIZER", "True")],
+            uses_baselines=False,
+            on_by_default=False,
+            default_test_len="short"
+        )
+
+###############################################################################
+class CSR(TestProperty):
+###############################################################################
+
+    def __init__(self, _):
+        TestProperty.__init__(
+            self,
+            "compute_santizer_racecheck",
+            "debug with compute sanitizer racecheck",
+            [("CMAKE_BUILD_TYPE", "Debug"),
+             ("EKAT_ENABLE_COMPUTE_SANITIZER", "True"),
+             ("EKAT_COMPUTE_SANITIZER_OPTIONS", "--tool racecheck")],
+            uses_baselines=False,
+            on_by_default=False,
+            default_test_len="short"
+        )
+
+###############################################################################
+class CSI(TestProperty):
+###############################################################################
+
+    def __init__(self, _):
+        TestProperty.__init__(
+            self,
+            "compute_santizer_initcheck",
+            "debug with compute sanitizer initcheck",
+            [("CMAKE_BUILD_TYPE", "Debug"),
+             ("EKAT_ENABLE_COMPUTE_SANITIZER", "True"),
+             ("EKAT_COMPUTE_SANITIZER_OPTIONS", "--tool initcheck")],
+            uses_baselines=False,
+            on_by_default=False,
+            default_test_len="short"
+        )
+
+###############################################################################
+class CSS(TestProperty):
```

Comment:
I'm guessing that compute-sanitizer does not allow 2+ tools to run at the same time, right?

Reply:
Yep, I specifically checked for that.